blogs/cert-manager-tls/yelb-gw.yaml

```diff
@@ -0,0 +1,150 @@
+apiVersion: v1
+kind: Namespace
+metadata:
+  name: yelb
+  labels:
+    appmesh.k8s.aws/sidecarInjectorWebhook: enabled
+    mesh: yelb
+    gateway: yelb-gw
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: VirtualGateway
+metadata:
+  name: yelb-gw
+  namespace: yelb
+spec:
+  backendDefaults:
+    clientPolicy:
+      tls:
+        enforce: true
+        validation:
+          trust:
+            file:
+              certificateChain: /etc/keys/yelb/ca.crt
+  namespaceSelector:
+    matchLabels:
+      gateway: yelb-gw
+  podSelector:
+    matchLabels:
+      app: yelb-gw
+  listeners:
+    - portMapping:
+        port: 8443
+        protocol: http
+      tls:
+        certificate:
+          file:
+            certificateChain: /etc/keys/yelb/tls.crt
+            privateKey: /etc/keys/yelb/tls.key
+        mode: STRICT        
+---
+apiVersion: appmesh.k8s.aws/v1beta2
+kind: GatewayRoute
+metadata:
+  name: gateway-route
+  namespace: yelb
+spec:
+  httpRoute:
+    match:
+      prefix: "/"
+    action:
+      target:
+        virtualService:
+          virtualServiceRef:
+            name: yelb-ui
+---
+apiVersion: cert-manager.io/v1alpha2
+kind: Certificate
+metadata:
+  name: yelb-cert-gw
+  namespace: yelb
+spec:
+  dnsNames:
+    - "yelb-gw.yelb.svc.cluster.local"
+  secretName: yelb-tls-gw
+  issuerRef:
+    name: ca-issuer
+---
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: yelb-gw
+  namespace: yelb
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      app: yelb-gw
+  template:
+    metadata:
+      labels:
+        app: yelb-gw
+    spec:
+      containers:
+        - name: envoy
+          image: 840364872350.dkr.ecr.us-west-2.amazonaws.com/aws-appmesh-envoy:v1.15.0.0-prod
```

Comment:
We prefer these are not hard-coded in the templates, and instead have folks reference https://docs.aws.amazon.com/app-mesh/latest/userguide/envoy.html for the latest image, using an ENV like `ENVOY_IMAGE` and a bash script to inject it to the right place.

The reason for this is: If v1.15 is found to have a CVE in the future, we'd need to come back and update each reference, otherwise folks are running a demo with a known security issue.